### PR TITLE
TRUNK-5951 : SerializedObject Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

replace hibernate xml mappings with annotations and removed relevant xml file.
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5951

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by ManojLL** [comment]
@dkayiwa Could you please review this PR?

File: api/src/main/java/org/openmrs/api/db/SerializedObject.java [hunk 1]
```diff
@@ -9,22 +9,45 @@
...
+	@GenericGenerator(
+		name = "serialized_object_id_seq",
+		strategy = "native",
+		parameters = @Parameter(name = "sequence", value = "serialized_object_serialized_object_id_seq")
+	)
+	@Column(name = "serialized_object_id")
 	private Integer id;
-	
+
+	@Column(name = "type", nullable = false)
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/SerializedObject.java:41]: Shouldn't these strings have a length attribute?

Comment (ManojLL) [reply api/src/main/java/org/openmrs/api/db/SerializedObject.java]: added string lengths

File: api/src/main/java/org/openmrs/api/db/SerializedObject.java [hunk 2]
```diff
@@ -9,22 +9,45 @@
...
 /**
  * Object representation of a Serialized Object as stored in the database.
  */
+@Entity
+@Table(name = "serialized_object")
 public class SerializedObject extends BaseChangeableOpenmrsMetadata {
-	
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "serialized_object_id_seq")
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/SerializedObject.java:32]: Is there any reason why you have not used `GenerationType.IDENTITY`?

Comment (ManojLL) [reply api/src/main/java/org/openmrs/api/db/SerializedObject.java:32]: @dkayiwa  The earlier implementation utilized GenerationType.SEQUENCE, and this choice is also reflected in [this pull request](https://github.com/openmrs/openmrs-core/pull/4388/files#72), which I used as a reference. Some of the code reviews suggested a modification from GenerationType.IDENTITY to GenerationType.SEQUENCE. 

 Please let me know, if I need to change `sequence` to `Identity`.

File: api/src/main/java/org/openmrs/api/db/SerializedObject.java (hunk 1 above)
Comment (ManojLL) [reply api/src/main/java/org/openmrs/api/db/SerializedObject.java]: @dkayiwa  any update on this?

File: api/src/main/java/org/openmrs/api/db/SerializedObject.java (hunk 2 above)
Comment (ManojLL) [reply api/src/main/java/org/openmrs/api/db/SerializedObject.java:32]: @dkayiwa  any update on this?

**Comment by dkayiwa** [comment]
@ManojLL can you look into these merge conflicts?

**Comment by ManojLL** [comment]
I have no idea how this PR got closed. This happen while I was resolving the merge conflicts.

**Pull request closed without merging**